R/ml_feature_one_hot_encoder.R

```diff
@@ -21,59 +22,130 @@ ml_one_hot_encoder <- ft_one_hot_encoder
 
 #' @export
 ft_one_hot_encoder.spark_connection <- function(x, input_col = NULL, output_col = NULL,
+                                                input_cols = NULL, output_cols = NULL, handle_invalid = "error",
                                                 drop_last = TRUE, uid = random_string("one_hot_encoder_"), ...) {
   .args <- list(
     input_col = input_col,
     output_col = output_col,
+    input_cols = input_cols,
+    output_cols = output_cols,
+    handle_invalid = handle_invalid,
     drop_last = drop_last,
     uid = uid
   ) %>%
     c(rlang::dots_list(...)) %>%
     validator_ml_one_hot_encoder()
+  version <- spark_version(x)
+  if(version < "3.0.0") {
+    jobj <- spark_pipeline_stage(
+      x, "org.apache.spark.ml.feature.OneHotEncoder",
+      input_col = .args[["input_col"]], output_col = .args[["output_col"]], uid = .args[["uid"]]
+    ) %>%
+      invoke("setDropLast", .args[["drop_last"]])
 
-  jobj <- spark_pipeline_stage(
-    x, "org.apache.spark.ml.feature.OneHotEncoder",
-    input_col = .args[["input_col"]], output_col = .args[["output_col"]], uid = .args[["uid"]]
-  ) %>%
-    invoke("setDropLast", .args[["drop_last"]])
+    estimator <- new_ml_one_hot_encoder(jobj)
+  } else {
+    estimator <- spark_pipeline_stage(
+      x, "org.apache.spark.ml.feature.OneHotEncoder",
+      input_col = .args[["input_col"]], output_col = .args[["output_col"]],
+      input_cols = .args[["input_cols"]], output_cols = .args[["output_cols"]], uid = .args[["uid"]]
+    ) %>%
+      invoke("setHandleInvalid", .args[["handle_invalid"]]) %>%
+      invoke("setDropLast", .args[["drop_last"]]) %>%
+      new_ml_one_hot_encoder_estimator()
+  }
 
-  new_ml_one_hot_encoder(jobj)
+  estimator
 }
 
 #' @export
 ft_one_hot_encoder.ml_pipeline <- function(x, input_col = NULL, output_col = NULL,
+                                           input_cols = NULL, output_cols = NULL, handle_invalid = "error",
                                            drop_last = TRUE, uid = random_string("one_hot_encoder_"), ...) {
-  stage <- ft_one_hot_encoder.spark_connection(
-    x = spark_connection(x),
-    input_col = input_col,
-    output_col = output_col,
-    drop_last = drop_last,
-    uid = uid,
-    ...
-  )
+  sc <- spark_connection(x)
```

Comment:
These three seem to be a useful combination. How about adding a new utility function like this:
```
is_spark_3 <- function() {
  version <- spark_connection() %>% version 
  return version >= "3"
}
```